### PR TITLE
Add arguments for inputs and output to the ConvertSequencingArtifactToOxoG command

### PR DESCRIPTION
**Pull request by pshapiro4broad**
### Description

The `ConvertSequencingArtifactToOxoG` command uses the `INPUT_BASE` argument to infer the paths of its input and output files. This reliance on inferring path locations makes it harder to integrate this tool with other tools that don't follow the same convention.

This change adds additional arguments for the two inputs and the command's output file. The new arguments are optional and the original command syntax will continue to work as before.

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [x] Edited the README / documentation (if applicable)
- [x] All tests passing on Travis

#### Review
- [x] Final thumbs-up from reviewer
- [x] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/analysis/artifacts/ConvertSequencingArtifactToOxoG.java [hunk 1]
```diff
@@ -26,48 +29,94 @@
...
+            "<a href='http://broadinstitute.github.io/picard/picard-metric-definitions.html#CollectOxoGMetrics.CpcgMetrics'>ConvertSequencingArtifactToOxoG</a> " +
+            "for detailed descriptions of the output metrics produced by this tool." +
+            "<hr />";
+
+    public static final String OXOG_METRICS_EXT = ".oxog_metrics";
+
     @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME,
-            doc = "Basename of the input artifact metrics file (output by CollectSequencingArtifactMetrics)")
+            doc = "Basename of the input artifact metrics file (output by CollectSequencingArtifactMetrics). If this is not " +
+                    " specified, you must specifcy PRE_ADAPTER_IN and BAIT_BIAS_IN",
```
Comment (gbggrant) [inline src/main/java/picard/analysis/artifacts/ConvertSequencingArtifactToOxoG.java:66]: "specify"

File: src/main/java/picard/analysis/artifacts/ConvertSequencingArtifactToOxoG.java [hunk 2]
```diff
@@ -26,48 +29,94 @@
...
 
+    @Argument(doc = "File for the output OxoG metrics. Defaults to a filename based on the output basename",
+            optional = true)
+    public File OXOG_OUT;
+
     @Override
-    protected int doWork() {
-        if (OUTPUT_BASE == null) { OUTPUT_BASE = INPUT_BASE; }
+    protected String[] customCommandLineValidation() {
+        List<String> errors = new ArrayList<>();
```
Comment (gbggrant) [inline src/main/java/picard/analysis/artifacts/ConvertSequencingArtifactToOxoG.java:89]: final

File: src/test/java/picard/analysis/artifacts/ConvertSequencingArtifactToOxoGTest.java [hunk 3]
```diff
@@ -60,74 +77,78 @@ public void testEquivalence() throws IOException, NoSuchFieldException {
...
-        Assert.assertTrue(areMetricsEqual(outputFileOxoG, new File(convertedArtifacts + ".oxog_metrics"), columnToCompare), "Metrics differ");
+        Assert.assertTrue(areMetricsEqual(outputFileOxoG, oxogMetricsOutput, COLUMNS_TO_COMPARE), "Metrics differ");
     }
 
     /**
      * Compare the metrics in two files, ignoring headers and histograms.
      */
     public static boolean areMetricsEqual(final File file1, final File file2, final String[] columnsToCompare) throws NoSuchFieldException {
-        try {
+        try (Reader reader1 = new FileReader(file1);
```
Comment (gbggrant) [inline src/test/java/picard/analysis/artifacts/ConvertSequencingArtifactToOxoGTest.java:103]: final, final

Comment (pshapiro4broad) [reply src/test/java/picard/analysis/artifacts/ConvertSequencingArtifactToOxoGTest.java:103]: They already are! Resources declared in a `try` are `final` by default. https://docs.oracle.com/javase/specs/jls/se8/html/jls-14.html#jls-14.20.3

**Pull request merged**